### ISSUE TITLE
Support XDG directories

### ISSUE THREAD

**Issue by gibfahn** [pulumi/pulumi#2534]
Simplest version would just be to read an env var like `$PULUMI_HOME` and use that instead of `$HOME/.pulumi` if defined. That way those that care can set the variable.

Something like what git does (see [the docs](https://git-scm.com/docs/git-config#FILES), it supports all the defaults, and env vars, and command line flags) would be the fancy version.

Sorry if this is the wrong place to raise this!

**Comment by ellismg** [comment]
> Sorry if this is the wrong place to raise this!

No, this is the correct place.  This is a change we'd be willing to accept, but would love help on, if you're interested.  Would be interesting to look at the XDG spec and see what parts of `~/.pulumi` would be mapped where (I could imagine that our plugin cache ends up under `$XDG_CACHE_HOME` or something.

**Comment by Selonophile** [comment]
I'd like to take this one up.

**Comment by Volatus** [comment]
This is actually a bigger change than I previously thought, there are a number of instances where the paths are hardcoded throughout the codebase. The easiest "fix" is to set the `$PULUMI_HOME` variable to the `$XDG_DATA_HOME/pulumi`. Yes, it doesn't put all files strictly where they belong but it's better than `$HOME/.pulumi`.

**Comment by thepkc** [comment]
Necromancing this: any luck? this is still relevant

**Comment by desuwulf** [comment]
@Volatus any chance for this:)?

**Comment by stooj** [comment]
It's not a fix, but you can set a `PULUMI_HOME` env var to override the `~/.pulumi` location.

Pulumi will still put everything in that directory, but at least it won't be in `$HOME`.
